tox_conda/plugin.py

```diff
@@ -1,332 +1,525 @@
-        # Append filenames of additional dependency sources. tox will automatically hash
-        # their contents to detect changes.
-        if envconfig.conda_spec is not None:
-            conda_deps.append(DepConfig(envconfig.conda_spec))
-        if envconfig.conda_env is not None:
-            conda_deps.append(DepConfig(envconfig.conda_env))
-        envconfig.deps.extend(conda_deps)
-
-        envconfig.conda_exe = conda_exe
-
-
-def find_conda():
-    # This should work if we're not already in an environment
-    conda_exe = os.environ.get("_CONDA_EXE")
-    if conda_exe:
-        return conda_exe
-
-    # This should work if we're in an active environment
-    conda_exe = os.environ.get("CONDA_EXE")
-    if conda_exe:
-        return conda_exe
-
-    path = shutil.which("conda")
-
-    if path is None:
-        _exit_on_missing_conda()
-
-    try:
-        subprocess.run([str(path), "-h"], stdout=subprocess.DEVNULL)
-    except subprocess.CalledProcessError:
-        _exit_on_missing_conda()
-
-    return path
-
-
-def _exit_on_missing_conda():
-    tox.reporter.error(MISSING_CONDA_ERROR)
-    raise SystemExit(0)
-
-
-def _run_conda_process(args, venv, action, cwd):
-    redirect = tox.reporter.verbosity() < tox.reporter.Verbosity.DEBUG
-    venv._pcall(args, venv=False, action=action, cwd=cwd, redirect=redirect)
-
-
-@hookimpl
-def tox_testenv_create(venv, action):
-    tox.venv.cleanup_for_venv(venv)
-    basepath = venv.path.dirpath()
+from tox.execute.api import Execute, ExecuteInstance, ExecuteOptions, ExecuteRequest, SyncWrite
+from tox.execute.local_sub_process import LocalSubProcessExecuteInstance, LocalSubProcessExecutor
+from tox.plugin import impl
+from tox.plugin.spec import EnvConfigSet, State, ToxEnvRegister
+from tox.tox_env.api import StdinSource, ToxEnvCreateArgs
+from tox.tox_env.errors import Fail
+from tox.tox_env.installer import Installer
+from tox.tox_env.python.api import PythonInfo, VersionInfo
+from tox.tox_env.python.pip.pip_install import Pip
+from tox.tox_env.python.pip.req_file import PythonDeps
+from tox.tox_env.python.runner import PythonRun
+
+__all__ = []
+
+
+class CondaEnvRunner(PythonRun):
+    # Can be overridden in tests
+    @staticmethod
+    def _default_execute_instance_factory(
+        request: ExecuteRequest, options: ExecuteOptions, out: SyncWrite, err: SyncWrite
+    ):
+        return LocalSubProcessExecuteInstance(request, options, out, err)
+
+    _execute_instance_factory: Callable = _default_execute_instance_factory
+
+    def __init__(self, create_args: ToxEnvCreateArgs) -> None:
+        self._installer = None
+        self._executor = None
+        self._external_executor = None
+        self._created = False
+        super().__init__(create_args)
+
+    @staticmethod
+    def id() -> str:  # noqa A003
+        return "conda"
+
+    def _get_python(self, base_python: List[str]) -> Optional[PythonInfo]:
+        exe_path = base_python[0]
+
+        output = self._run_pure(
```

Comment:
Shold be cached to do fresh run every time 🤔 something like virtualenv does

Reply:
Do you mean we should run this only once, and cache the result?

Reply:
Yeah something like https://github.com/pypa/virtualenv/blob/main/src/virtualenv/discovery/cached_py_info.py#LL28C1-L28C1

Reply:
Great idea, I will do it.